addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -95,6 +102,21 @@ def _compute_state(self):
             elif all(picking.state in ['cancel', 'done'] for picking in batch.picking_ids):
                 batch.state = 'done'
 
+    @api.depends('picking_ids.scheduled_date')
+    def _compute_scheduled_date(self):
+        """ scheduled_date = the earliest scheduled_date across all pickings. """
+        for batch in self:
+            dates_list = batch.picking_ids.filtered(lambda p: p.scheduled_date).mapped('scheduled_date')
+            if dates_list:
+                batch.scheduled_date = fields.Datetime.to_string(min(dates_list))
+            else:
+                batch.scheduled_date = False
```

Comment:
The `filtered` isn't necessary as pickings will always have a `scheduled_date`.
Also, you can set `False` by default on all the records at once:
```python
self.scheduled_date = False
for batch in self:
    dates_list = batch.picking_ids.mapped('scheduled_date')
    if dates_list:
        batch.scheduled_date = fields.Datetime.to_string(min(dates_list))
```

Reply:
Good to know for both, thanks! :)